modules/proxy/proxy_util.c

```diff
@@ -2768,7 +2768,16 @@ ap_proxy_determine_connection(apr_pool_t *p, request_rec *r,
             socket_cleanup(conn);
             conn->close = 0;
         }
-        if (will_reuse) {
+        if (will_reuse) {         
+            if(!strcmp(conn->hostname, worker->s->hostname_ex) || conn->port != worker->s->port) {
+                apr_sockaddr_t *addr;
+                err = apr_sockaddr_info_get(&addr,
+                                            worker->s->hostname_ex, APR_UNSPEC,
+                                            worker->s->port, 0,
+                                            worker->cp->dns_pool);
+                worker->cp->addr = addr;
+            }
```

Comment:
Coincidentaly I just started a discussion there https://lists.apache.org/thread/jftfd6mo6p3b0tw694rlh09gdl7189hq about the unsafety of changing worker->cp->addr like this. We need a thread-safe way to do it, probably outside the scope of this PR though..

Reply:
Not to talk about changing worker->s->hostname_ex/port (even under the worker lock) while there are concurrent readers like here which don't take the lock.